amf-graphql/shared/src/main/scala/amf/graphql/internal/spec/parser/syntax/GraphQLASTParserHelper.scala

```diff
@@ -47,11 +47,8 @@ trait GraphQLASTParserHelper extends AntlrASTParserHelper {
     val maybeName = potentialPaths.map { p =>
       path(n, p)
     } collectFirst {
-      case Some(t: Terminal) => t.value
-      case Some(n: Node) =>
-        n.children match {
-          case mutable.Buffer(onlyChild: Terminal) => onlyChild.value
-        }
+      case Some(t: Terminal)                     => t.value
+      case Some(n: Node) if n.children.size == 1 => n.children.head.asInstanceOf[Terminal].value
```

Comment:
is the first value of the nodes children always a terminal ??

Reply:
in the case of the paths that may have a name, they all do. But that's only the case of a semi-correct API, so I've added an extra check in the match just in case 👌🏽